storage/rocksdb/rdb_datadic.h

```diff
@@ -42,6 +42,48 @@ class Rdb_field_packing;
 class Rdb_cf_manager;
 class Rdb_ddl_manager;
 
+class Rdb_merge_operator : public rocksdb::AssociativeMergeOperator {
+ public:
+  virtual bool Merge(const rocksdb::Slice &key,
+                     const rocksdb::Slice *existing_value,
+                     const rocksdb::Slice &value, std::string *new_value,
+                     rocksdb::Logger *logger) const override {
+    
+    DBUG_ASSERT(new_value != nullptr);
+    std::string merged_value;
+
+    /* if no existing vaue, then persist the passed in value */
+    if (existing_value == nullptr) {
+      merged_value = Serialize(value);
+    } else {
+      merged_value = Deserialize(std::max(Serialize(value), Serialize(*existing_value)));
```

Comment:
Serialize typically means encode for storing on media/networking. Deserialize typically means decode, so I think you have the opposite here meaning here.

Also, this code looks wrong? In one case, merged_value is the product of a Serialize() call. In the second case, it is the product of a Deserialize() call.